runtime/doc/lsp.txt

```diff
@@ -697,9 +696,8 @@ buf_request_sync({bufnr}, {method}, {params}, {timeout_ms})
                       result. Defaults to 1000
 
     Return: ~
-        table<integer, any>|nil result, string|nil err Map of
-        client_id:request_result. On timeout, cancel or error, returns `(nil,
-        err)` where `err` is a string describing the failure reason.
+        table<integer, {err: lsp.ResponseError , result: any}>|nil result, string|nil err Map of
+        client_id:request_result. On timeout, cancel or error, returns `(nil, err)` where `err` is a string describing the failure reason.
```

Comment:
Looks like this causes failures with the vimdoc parser:

```
FAILED   test/functional/lua/help_spec.lua @ 13: :help docs validate
test/functional/lua/help_spec.lua:25: expected no parse errors, got: 1

stack traceback:
	test/functional/lua/help_spec.lua:25: in function <test/functional/lua/help_spec.lua:13>
```

It doesn't like the `|` 

Using `:InspectTree` on the help file shows the `ERROR` node:

![image](https://user-images.githubusercontent.com/38700/224469872-ab3bffd0-b6d8-44a4-bf50-7b8d77548271.png)

Reply:
Exactly. Try separating the `|` with spaces (which is not legal for taglinks, which this conflicts with).

If this becomes too cumbersome, we could switch to `foo?` for optional parameters.

Reply:
Fixed as suggested by folke. The remaining test failures don't seem to be related to this.